src/main/java/springboot/rest/specifications/CustomSpecifications.java

```diff
@@ -216,14 +228,26 @@ else if (join == null) {
                 return builder.equal(root.get(a.getName()), val);
             }
             else if (a.isAssociation()) {
-                return prepareJoinAssociatedPredicate(root, a, val);
+                return prepareJoinAssociatedPredicate(builder, root, a, val);
+            } else if(isSerializableAndFromString(a)) {
+                try {
+                    return builder.equal (root.get(a.getName()), a.getJavaType().getMethod("fromString", String.class).invoke(null, val.toString()));
+                } catch (IllegalAccessException | InvocationTargetException | NoSuchMethodException e) {
+                    throw new IllegalArgumentException("equality/inequality is currently supported on primitives, enums and serializables with string constructor", e);
+                }
             }
         }
         else if (join != null) {
             if (isEnum(a)) {
                 return builder.equal(join.get(a.getName()), Enum.valueOf(Class.class.cast(a.getJavaType()), (String) val));
             } else if (isPrimitive(a)) {
                 return builder.equal(join.get(a.getName()), val);
+            } else if(isSerializableAndFromString(a)) {
+                try {
+                    return builder.equal (join.get(a.getName()), a.getJavaType().getMethod("fromString", String.class).invoke(null, val.toString()));
+                } catch (IllegalAccessException | InvocationTargetException | NoSuchMethodException e) {
+                    throw new IllegalArgumentException("equality/inequality is currently supported on primitives, enums and serializables with string constructor", e);
```

Comment:
those are identical need to be extracted

Reply:
Please recheck :)